<?xml version="1.0"?>
<launch>
  <arg name="robot_ip" default="192.168.1.102"/>
  

  <rosparam file="$(find soma_ur5)/config/ur5_settings.yaml" command="load"/>
  <param name="use_sim_time" value="false" unless="$(arg simulated)" />
  <param name="use_sim_time" value="true" if="$(arg simulated)" />




  <!-- send robot urdf to param server -->
  <include if="$(arg simulated)" file="$(find soma_ur5)/launch/ur5_gazebo.launch"/>
  <include if="$(arg simulated)" file="$(find soma_ur5)/launch/ur5_control_gazebo.launch"/>
  <node if="$(arg simulated)" pkg="soma_ur5" type="gazebo_bridge" name="gazebo_bridge"/>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  
  <include unless="$(arg simulated)" file="$(find ur_modern_driver)/launch/ur5_bringup_joint_limited.launch">
    <arg name="robot_ip" value="$(arg robot_ip)"/>
  </include>
  <include unless="$(arg simulated)" file="$(find qb_interface)/launch/qb_interface.launch"/>

  <node pkg="soma_ur5" type="pisa_hand_bridge" name="pisa_hand_bridge"/>
  <node unless="$(arg simulated)" pkg="soma_ur5" type="controller" name="controller"/>
  <!--node pkg="soma_ur5" type="haptic" name="haptic" /-->


 <node unless="$(arg simulated)" pkg="netft_rdt_driver" type="netft_node" name="netft_node" args="--address 192.168.1.1" />
 <node unless="$(arg simulated)" pkg="soma_ur5" type="ati_bridge" name="ati_bridge" />
  
</launch>
